.github/workflows/updater.sh

```diff
@@ -0,0 +1,136 @@
+echo "Latest release from upstream: $version"
+echo "VERSION=$version" >> $GITHUB_ENV
+echo "REPO=$repo" >> $GITHUB_ENV
+# For the time being, let's assume the script will fail
+echo "PROCEED=false" >> $GITHUB_ENV
+
+# Proceed only if the retrieved version is greater than the current one
+if [[ ! "$version" > "$current_version" ]] ; then
+    echo "::warning ::No new version available"
+    exit 0
+# Proceed only if a PR for this new version does not already exist
+elif git ls-remote -q --exit-code --heads https://github.com/$GITHUB_REPOSITORY.git ci-auto-update-v$version ; then
+    echo "::warning ::A branch already exists for this update"
+    exit 0
+fi
+
+#=================================================
+# UPDATE SOURCE FILES
+#=================================================
+
+# Here we use the $assets variable to get the resources published in the upstream release.
+# Here is an example for Grav, it has to be adapted in accordance with how the upstream releases look like.
+
+# Let's loop over the array of assets URLs
+for src in mc_amd64 mc_arm64 amd64 arm64; do
+
+case $src in
+  "amd64")
+    asset_url="https://dl.min.io/server/minio/release/linux-amd64/archive/minio.$version"
+    src_filename=minio
+    ;;
+  "arm64")
+    asset_url="https://dl.min.io/server/minio/release/linux-arm64/archive/minio.$version"
+    src_filename=minio
+    ;;
+  "mc_amd64")
+    asset_url="https://dl.min.io/client/mc/release/linux-arm64/archive/mc.$mc_version"
+    src_filename=mc
+    ;;
+  "mc_arm64")
+    asset_url="https://dl.min.io/client/mc/release/linux-arm64/archive/mc.$mc_version"
+    src_filename=mc
+    ;;
+esac
+
+echo "Handling asset at $asset_url"
+
+
+# If $src is not empty, let's process the asset
+if [ ! -z "$src" ]; then
+
+# Create the temporary directory
+tempdir="$(mktemp -d)"
+
+# Download sources and calculate checksum
+filename=${asset_url##*/}
+curl --silent -4 -L $asset_url -o "$tempdir/$filename"
+checksum=$(sha256sum "$tempdir/$filename" | head -c 64)
+
+# Delete temporary directory
+rm -rf $tempdir
+
+# Get extension
+if [[ $filename == *.tar.gz ]]; then
+  extension=tar.gz
+else
+  extension=${filename##*.}
+fi
+
+# Rewrite source file
+cat <<EOT > conf/$src.src
+SOURCE_URL=$asset_url
+SOURCE_SUM=$checksum
+SOURCE_SUM_PRG=sha256sum
+SOURCE_IN_SUBDIR=false
+SOURCE_FILENAME=$src_filename
+SOURCE_EXTRACT=false
+EOT
+echo "... conf/$src.src updated"
+
+else
+echo "... asset ignored"
+fi
+
+done
+
+#=================================================
+# SPECIFIC UPDATE STEPS
+#=================================================
+
+# Any action on the app's source code can be done.
+# The GitHub Action workflow takes care of committing all changes after this script ends.
+
+#=================================================
+# GENERIC FINALIZATION
+#=================================================
+
+# Replace new version in manifest
+echo "$(jq -s --indent 4 ".[] | .version = \"$version~ynh1\"" manifest.json)" > manifest.json
```

Comment:
```suggestion
manifestversion=$(sed -E "s/RELEASE\.([0-9-]+)T.+/\1/g" <<< $version | sed -E "s/-/./g")
echo "$(jq -s --indent 4 ".[] | .version | sub("RELEASE.";"") | split("T")[0] = \"$manifestversion~ynh1\"" manifest.json)" > manifest.json
```

Cleans up the version for the manifest.